eval/src/apps/tensor_conformance/generate.cpp

```diff
@@ -131,43 +146,74 @@ void generate_op2_join(const vespalib::string &op2_expr, const Sequence &seq, Te
+    generate_op2_join("a<=b", Div16(N()), dst);
+    generate_op2_join("a>b", Div16(N()), dst);
+    generate_op2_join("a>=b", Div16(N()), dst);
     generate_op2_join("a&&b", Mask2Seq(SkipNth(3)), dst);
     generate_op2_join("a||b", Mask2Seq(SkipNth(3)), dst);
-    generate_op2_join("atan2(a,b)", Div10(N()), dst);
-    generate_op2_join("ldexp(a,b)", Div10(N()), dst);
-    generate_op2_join("fmod(a,b)", Div10(N()), dst);
-    generate_op2_join("min(a,b)", Div10(N()), dst);
-    generate_op2_join("max(a,b)", Div10(N()), dst);
-    generate_join_expr("join(a,b,f(a,b)((a+b)/(a*b)))", Div10(N()), dst);
+    generate_op2_join("atan2(a,b)", Div16(N()), dst);
+    generate_op2_join("ldexp(a,b)", Div16(N()), dst);
+    generate_op2_join("fmod(a,b)", Div16(N()), dst);
+    generate_op2_join("min(a,b)", Div16(N()), dst);
+    generate_op2_join("max(a,b)", Div16(N()), dst);
+    generate_join_expr("join(a,b,f(a,b)((a+b)/(a*b)))", Div16(N()), dst);
 }
 
 //-----------------------------------------------------------------------------
 
+void generate_dot_product(TestBuilder &dst,
+                          double expect,
+                          const Layout &lhs, const Seq &lhs_seq,
+                          const Layout &rhs, const Seq &rhs_seq)
+{
+    dst.add("reduce(a*b,sum)",
+            { {"a", spec(lhs, lhs_seq)},{"b", spec(rhs, rhs_seq)} },
+            spec(expect));
+}
+
+void generate_dot_product(TestBuilder &dst,
+                          double expect,
+                          const Layout &layout,
+                          const Seq &lhs_seq,
+                          const Seq &rhs_seq)
+{
+    auto fl_lay = float_cells(layout);
+    generate_dot_product(dst, expect, layout, lhs_seq, layout, rhs_seq);
+    generate_dot_product(dst, expect, fl_lay, lhs_seq, layout, rhs_seq);
+    generate_dot_product(dst, expect, layout, lhs_seq, fl_lay, rhs_seq);
+    generate_dot_product(dst, expect, fl_lay, lhs_seq, fl_lay, rhs_seq);
+}
+
 void generate_dot_product(TestBuilder &dst) {
-    dst.add("reduce(a*b,sum)", {{"a", spec(x(3), Seq({ 2, 3, 5 }))}, {"b", spec(x(3), Seq({ 7, 11, 13 }))}},
-            spec((2 * 7) + (3 * 11) + (5 * 13)));
+    generate_dot_product(dst, ((2 * 7) + (3 * 11) + (5 * 13)), {x(3)},
+                         Seq({ 2, 3, 5 }),
+                         Seq({ 7, 11, 13 }));
 }
 
 //-----------------------------------------------------------------------------
 
 void generate_xw_product(TestBuilder &dst) {
     auto matrix = spec({x(2),y(3)}, Seq({ 3, 5, 7, 11, 13, 17 }));
+    auto fmatrix = spec(float_cells({x(2),y(3)}), Seq({ 3, 5, 7, 11, 13, 17 }));
     dst.add("reduce(a*b,sum,x)", {{"a", spec(x(2), Seq({ 1, 2 }))}, {"b", matrix}},
             spec(y(3), Seq({(1*3+2*11),(1*5+2*13),(1*7+2*17)})));
+    dst.add("reduce(a*b,sum,x)",
+            {{"a", spec(float_cells({x(2)}), Seq({ 1, 2 }))}, {"b", matrix}},
+            spec(y(3), Seq({(1*3+2*11),(1*5+2*13),(1*7+2*17)})));
+    dst.add("reduce(a*b,sum,x)",
```

Comment:
consider adding test for float matrix and double vector

Reply:
added